src/M6Web/Bundle/ElasticsearchBundle/Connection/GuzzleConnectionDecorator.php

```diff
@@ -71,10 +75,34 @@ protected function dispatchEvent($method, $uri, $statusCode, $duration)
                 ->setUri($uri)
                 ->setMethod($method)
                 ->setStatusCode($statusCode)
-                ->setDuration($duration);
+                ->setDuration($duration)
+                ->setTook($took);
 
             $this->eventDispatcher->dispatch('m6web.elasticsearch', $event);
         }
     }
 
-}
\ No newline at end of file
+    /**
+     * Extract took from response
+     *
+     * @param array $response
+     *
+     * @return int|null
+     */
+    protected function extractTookFromResponse(array $response)
+    {
+        if (!isset($response['text'])) {
+
+            return null;
+        }
+
+        $result = json_decode($response['text'], true);
```

Comment:
I think we should avoid to do a full `json_decode()` here. For big resultSets this can lead to a huge memory consumption.

May be we can use `json_decode()` with a limited `depth` parameter, it might be more memory-friendly (not sure about that).

Or we can get the value with a regex...